make.js

```diff
@@ -1,14 +1,36 @@
 require('shelljs/make');
 var path = require('path');
 var fs = require('fs');
+var semver = require('semver');
+var ncp = require('child_process');
 
 var rp = function (relPath) {
     return path.join(__dirname, relPath);
 }
 
+var fail = function (message) {
+    console.error('ERROR: ' + message);
+    process.exit(1);
+}
+
 var buildPath = path.join(__dirname, '_build');
 var testPath = path.join(__dirname, 'test');
 
+// enforce minimum Node version
+var minimumNodeVersion = '6.12.0';
```

Comment:
I tested and this works with lts/argon -> v4.8.6.  lets make that the minimum.  Since we have to test that it runs on argon, it's convenient to dev on it (use nvm and then build, test, samples.  Alternatively, we could keep 6.12 as dev minimum but then to test on argon you would have to build on 6.12 and then nvm to 4.x and then run tests.

Reply:
It seems worthwhile to update our documentation to say what node versions you can run against. 

I was also thinking of adding a make.js step for that automates what you just said(using nvm), building and then running tests across all supported node versions. Of course this assumes that nvm is installed and setup correctly(we could also add a courtesy nvm install for missing versions) but it seems useful to have this ability during development to say run allversiontests (or something along these lines) and know you are getting full testing before pushing.

What do you think?

Reply:
Yeah, I was thinking the same on both fronts.  Go ahead.

Reply:
Great, I created new tickets for to track them.